share/regulate/steps/__init__.py

```diff
@@ -21,8 +23,14 @@ def error(self, description, node_id=None, exception=None):
         self.logs.append(log)
         raise RegulateError('Regulation failed: {}'.format(description)) from exception
 
+    @abc.abstractmethod
+    def run(self, graph):
+        pass
```

Comment:
I have very little faith in `abc`, so adding `raise NotImplementedError` makes me feel a little better.